scheduler-3/src/test/scala/uk/sky/scheduler/circe/CirceSerdesSpec.scala

```diff
@@ -11,10 +14,27 @@ import uk.sky.scheduler.util.Generator.given
 
 final class CirceSerdesSpec extends AnyWordSpec, ScalaCheckPropertyChecks, Matchers, EitherValues {
 
+  final case class JsonScheduleWithoutHeaders(
+      time: Long,
+      topic: String,
+      key: String,
+      value: Option[String]
+  )
+
+  given scheduleEncoderWithoutHeaders: Encoder[JsonScheduleWithoutHeaders] =
+    semiauto.deriveEncoder[JsonScheduleWithoutHeaders]
+
+  given Arbitrary[JsonScheduleWithoutHeaders] = Arbitrary(Gen.resultOf(JsonScheduleWithoutHeaders.apply))
+
   "jsonSchedulerDecoder" should {
     "decode and encode a JSON schedule" in forAll { (schedule: JsonSchedule) =>
       decode[JsonSchedule](schedule.asJson.noSpaces).value shouldBe schedule
     }
+
+    "decode and encode a JSON schedule without headers" in forAll { (schedule: JsonScheduleWithoutHeaders) =>
```

Comment:
For consistency with the other [test scenario](https://github.com/sky-uk/kafka-message-scheduler/pull/392/files#diff-a6326e653cc59a0f0fb11d85c48064faf8b8c255c517c0875989f61835e3b201R54) that was renamed...

```suggestion
    "decode and encode a JSON schedule with empty headers" in forAll { (schedule: JsonScheduleWithoutHeaders) =>
```

Reply:
A schedule without headers and with empty headers are two different things in the context of these tests, so I would rather keep this test definition.